src/Microsoft.AspNet.Razor.Runtime/TagHelpers/DefaultTagHelperContent.cs

```diff
@@ -0,0 +1,146 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.Framework.Internal;
+
+namespace Microsoft.AspNet.Razor.Runtime.TagHelpers
+{
+    /// <summary>
+    /// Default concrete <see cref="TagHelperContent"/>.
+    /// </summary>
+    public class DefaultTagHelperContent : TagHelperContent, ITextWriterCopyable
+    {
+        private readonly BufferEntryCollection _buffer;
+
+        /// <summary>
+        /// Instantiates a new instance of <see cref="DefaultTagHelperContent"/>.
+        /// </summary>
+        public DefaultTagHelperContent()
+        {
+            _buffer = new BufferEntryCollection();
+        }
+
+        /// <inheritdoc />
+        public override bool IsModified
+        {
+            get
+            {
+                return _buffer.IsModified;
+            }
+        }
+
+        /// <inheritdoc />
+        public override bool IsWhiteSpace
+        {
+            get
+            {
+                foreach (var value in _buffer)
+                {
+                    if (!string.IsNullOrWhiteSpace(value))
+                    {
+                        return false;
+                    }
+                }
+
+                return true;
+            }
+        }
+
+        /// <inheritdoc />
+        public override bool IsEmpty
+        {
+            get
+            {
+                foreach (var value in _buffer)
+                {
+                    if (!string.IsNullOrEmpty(value))
+                    {
+                        return false;
+                    }
+                }
+
+                return true;
```

Comment:
nit: given changes to `Append()`, fact `_buffer` is `private`, and `_buffer` providing an enumerator, should only need to `return !_buffer.Any();`